commit 3f7759e7b7585a0bffda06d4eddc6b0b850ef6c3
Author: Iskren Chernev <me@iskren.info>
Date:   Wed Dec 30 17:29:43 2020 +0200

    drm/msm: Add modparam to allow vram carveout
    
    Using the GPU with a VRAM Carveout is a security vulnerability.
    Nevertheless it is sometimes required, especially when no IOMMU
    implementation is available for a certain platform.
    
    Signed-off-by: Iskren Chernev <iskren.chernev@gmail.com>
    Signed-off-by: Rob Clark <robdclark@chromium.org>

diff --git a/drivers/gpu/drm/msm/adreno/a3xx_gpu.c b/drivers/gpu/drm/msm/adreno/a3xx_gpu.c
index 93da6683a866..4534633fe7cd 100644
--- a/drivers/gpu/drm/msm/adreno/a3xx_gpu.c
+++ b/drivers/gpu/drm/msm/adreno/a3xx_gpu.c
@@ -564,8 +564,10 @@ struct msm_gpu *a3xx_gpu_init(struct drm_device *dev)
 		 * implement a cmdstream validator.
 		 */
 		DRM_DEV_ERROR(dev->dev, "No memory protection without IOMMU\n");
-		ret = -ENXIO;
-		goto fail;
+		if (!allow_vram_carveout) {
+			ret = -ENXIO;
+			goto fail;
+		}
 	}
 
 	icc_path = devm_of_icc_get(&pdev->dev, "gfx-mem");